controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -249,6 +260,24 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// ClusterRegistryConfig represents registry config used with the cluster.
 	// +optional
 	ClusterRegistryConfig *RegistryConfig `json:"clusterRegistryConfig,omitempty"`
+
+	// AutoNode set the AutoNode mode and AutoNode role ARN.
```

Comment:
Since this will be exposed as a YAML field, the doc should start with `autoNodeMode`, rather than the usual Go convention.